src/main/java/pl/scoutCamp/business/TeamCategorizationSheetService.java

```diff
@@ -27,15 +31,50 @@ public List<TeamCategorizationSheet> findTeamsSheetsByPeriod(String periodName)
     }
 
     public List<TeamCategorizationSheet> findTeamsSheetsByRegimentAndPeriod(Integer regimentId, String periodName) {
-        List<TeamCategorizationSheet> AllTeamsSheetsInRegiment =
-                teamCategorizationSheetDAO.findTeamsSheetsByPeriod(periodName)
+        List<TeamCategorizationSheet> allTeamsSheetsInRegiment =
+                findTeamsSheetsByPeriod(periodName)
                         .stream()
                         .filter(sheet -> sheet.getTeam().getRegiment().getId().equals(regimentId))
                         .toList();
-        if (AllTeamsSheetsInRegiment.isEmpty()) {
+        if (allTeamsSheetsInRegiment.isEmpty()) {
             throw new NotFoundException("Could not find list of teams' scores in this regiment");
         }
-        log.info("Number of teams' scores: [{}] in regiment with Id: [{}] in period: [{}]", AllTeamsSheetsInRegiment.size(), regimentId, periodName);
-        return AllTeamsSheetsInRegiment;
+        log.info("Number of teams' scores: [{}] in regiment with Id: [{}] in period: [{}]", allTeamsSheetsInRegiment.size(), regimentId, periodName);
+        return allTeamsSheetsInRegiment;
+    }
+
+    public List<TeamCategorizationSheet> getSortedRanking(String periodName, String sortedBy, String order) {
+        ArrayList<TeamCategorizationSheet> rankingList =
+                new ArrayList<>(findTeamsSheetsByPeriod(periodName));
+        return sortRanking(rankingList, sortedBy, order);
+    }
+
+    public List<TeamCategorizationSheet> getSortedRanking(Integer regimentId, String periodName, String sortedBy, String order) {
+        ArrayList<TeamCategorizationSheet> rankingList =
+                new ArrayList<>(findTeamsSheetsByRegimentAndPeriod(regimentId, periodName));
+        return sortRanking(rankingList, sortedBy, order);
+    }
+
+    public List<TeamCategorizationSheet> sortRanking(List<TeamCategorizationSheet> rankingToSort, String sortedBy, String order) {
+        if (order.equals("asc")) {
```

Comment:
I think it can be simplified. You can avoid writing sorting two times for each case in "if" and then again in "else". You can write sorting in switch at the beginning of the method and reverse the order then if (order == "desc"). WDYT?